packages/next/pages/_document.tsx

```diff
@@ -371,7 +371,12 @@ export class Main extends Component {
   render() {
     const { amphtml, html } = this.context._documentProps
     if (amphtml) return '__NEXT_AMP_RENDER_TARGET__'
-    return <div id="__next" dangerouslySetInnerHTML={{ __html: html }} />
+    return (
+      <>
+        <div id="__next" dangerouslySetInnerHTML={{ __html: html }} />
+        {process.env.NODE_ENV === 'development' && <div id="__next-build-watcher" />}
```

Comment:
I meant `document.createElement` etc in the client-side js, there's no need to have it right here.